library/src/engine/engine.ts

```diff
@@ -875,15 +875,15 @@ function applyAttributePlugin(
   attrKey: string,
   value: string,
 ): void {
-  const rawKey = camel(alias ? attrKey.slice(alias.length) : attrKey)
+  const rawKey = alias ? lowerFirst(attrKey.slice(camel(alias).length)) : attrKey
```

Comment:
This was calling camel just to lowercase the first letter. Calling camel messed things up when there were literal dashes in the key though. Also, it was only needed in the case that we have an alias. Called camel here on the alias as well in case there was a dash in there the length will have changed.

Reply:
On second thought the alias should probably just be camelized once when it's set rather than every time we are matching here.